docs/code_style.md

```diff
@@ -35,7 +35,7 @@ pip install -e ".[lint,mypy]"
     Check all application and test code with:
 
     ```sh
-    flake8 synapse tests
+    flake8 .
```

Comment:
That's right. Motivation here: https://github.com/matrix-org/synapse/issues/12107

In a nutshell: I think we should be configuring linters to select sensible files, rather than propagating lists of directories around everywhere.

This isn't specific to poetry, but I've snuck it in here because I'm making use of this action: https://github.com/matrix-org/backend-meta/blob/main/.github/workflows/python-poetry-ci.yml#L19-L25 which doesn't accept a list of targets for for the reason described in the previous sentence.

Reply:
Makes sense, thanks for explaining!